presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/source/AbstractJdbiBenchmarkSuiteSupplier.java

```diff
@@ -40,11 +39,8 @@ public BenchmarkSuite get()
     {
         return jdbi.inTransaction(handle -> {
             BenchmarkSuiteDao dao = handle.attach(BenchmarkSuiteDao.class);
-            BenchmarkSuiteInfo suiteInfo = dao.getBenchmarkSuiteInfo(suitesTableName, suite);
-            return new BenchmarkSuite(
-                    suite,
-                    suiteInfo,
-                    dao.getBenchmarkQueries(queriesTableName, suiteInfo.getQuerySet()));
+            BenchmarkSuite.JdbiBuilder suiteBuilder = dao.getBenchmarkSuite(suitesTableName, suite);
+            return suiteBuilder.setQueries(dao.getBenchmarkQueries(queriesTableName, suiteBuilder.getQuerySet())).build();
```

Comment:
You can find that all fields of `BenchmarkSuite` are immutable. `queries` is mutable only in this `JdbiBuilder` inner-class.

> Why do we need to use setQueries instead of make it part of the constructor?
This is because we need to run 2 MySQL queries to fetch all information about a suite.
1. `dao.getBenchmarkSuite` gets some the suite configuration by reading `benchmark_suites`.
2. `dao.getBenchmarkQueries` gets the list of queries by reading `benchmark_queries.`

The result of 1st query needs to be stored in some intermediate object. Before the refactoring, the result is mapped to `BenchmarkSuiteInfo` class, and a `BenchmarkSuite` contains both `BenchmarkSuiteInfo` and the query lists. With the refactoring, I removed `BenchmarkSuiteInfo`, and replace it with the builder class instead. The result of the 1st query is mapped to `BenchmarkSuite.JdbiBuilder` using the constructor, while the result of the 2nd query is added to the build using the setter. And finally `JdbiBuilder.build()` returns a `BenchmarkSuite` object.